src/compiler/utilities.ts

```diff
@@ -2293,12 +2293,12 @@ namespace ts {
             if (options.jsx === JsxEmit.Preserve) {
                 if (isSourceFileJavaScript(sourceFile)) {
                     if (fileExtensionIs(sourceFile.fileName, ".jsx")) {
-                        extension = ".jsx";
+                        extension = options.jsxExtName || ".jsx";

```

Comment:
That'd probably be the most user-friendly thing, though @mhegazy could answer better.